c10/core/TensorImpl.h

```diff
@@ -1554,29 +1554,41 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
         const U,
         U>;
 
+    template <typename DataPtrImplAccessor>
     MaybeConstT<T>* operator()(
         Self& self,
-        MaybeConstT<void>* (Storage::*data_accessor)() const) {
+        const DataPtrImplAccessor& data_ptr_impl_accessor) {
       TORCH_CHECK(
           self.data_type_.template Match<T>(),
           "Tensor type mismatch, caller expects elements to be ",
           caffe2::TypeMeta::TypeName<T>(),
           ", while tensor contains ",
           self.data_type_.name(),
           ". ");
-      return data_ptr_impl<T, Self>()(self, data_accessor);
+      return (self.*data_ptr_impl_accessor)();
     }
   };
 
  public:
+  /**
+   * More efficient helper for Tensor::data_ptr(). Like data<T>(), but
+   * does not do a type check. Unlike the untemplated data(), does
+   * check has_storage() and storage_initialized().
+   */
```

Comment:
If this is an internal helper, why is it public?

Reply:
Tensor::data_ptr. We can make it private and friend TensorBase if we want.

Reply:
OK. I guess it would have been better if we called this fast or unsafe, but nbd.

Reply:
it's trivial to rename, that can be a separate change altogether.